drivers/display/display_st7789h2.c

```diff
@@ -0,0 +1,645 @@
+	struct st7789h2_data *data = dev->data;
+	uint8_t shift = 0;
+	uint8_t offset = 0;
+	gpio_port_value_t val = 0;
+	gpio_port_pins_t mask = 0;
+	gpio_port_pins_t valmask = 0;
+
+	for (uint8_t i = 0; i < data->port_groups_count; i++) {
+
+		/* Get the bits of the value*/
+		shift = data->port_groups[i].len;
+		offset = data->port_groups[i].offset;
+
+		valmask = ((1<<shift) - 1);
+
+		val = ((value) & valmask) << offset;
+
+		mask = valmask << offset;
+
+		gpio_port_set_masked(data->port_groups[i].port, mask, val);
+
+		value >>= shift;
+	}
+#else
+	const struct st7789h2_config *config = dev->config;
+
+	for (uint8_t i = 0; i < config->ndata_gpios; i += 4) {
+		gpio_pin_set_dt(&(config->data_gpios[i]), value&0x1);
+		gpio_pin_set_dt(&(config->data_gpios[i+1]), value&0x2);
+		gpio_pin_set_dt(&(config->data_gpios[i+2]), value&0x4);
+		gpio_pin_set_dt(&(config->data_gpios[i+3]), value&0x8);
+		value >>= 4;
+	}
+#endif
+}
+
+static inline void st7789h2_wrn_pulse(const struct st7789h2_config *config)
+{
+	gpio_pin_set_dt(&config->wrn_gpio, 0);
+	gpio_pin_set_dt(&config->wrn_gpio, 1);
+}
+
+static void st7789h2_set_lcd_margins(struct st7789h2_data *data,
+			     uint16_t x_offset, uint16_t y_offset)
+{
+	data->x_offset = x_offset;
+	data->y_offset = y_offset;
+}
+
+static inline void st7789h2_set_cmd(const struct st7789h2_config *config, int is_cmd)
+{
+	gpio_pin_set_dt(&config->cmd_data_gpio, is_cmd);
+}
+
+static inline void st7789h2_set_cs(const struct st7789h2_config *config, int select)
+{
+	gpio_pin_set_dt(&config->cs_gpio, select);
+}
+
+static void st7789h2_transmit(const struct device *dev, uint8_t cmd,
+		uint8_t *tx_data, size_t tx_count)
+{
+	const struct st7789h2_config *config = dev->config;
+
+	st7789h2_set_cs(config, 1);
+	st7789h2_set_cmd(config, 1);
+
+	st7789h2_data_bus_set(dev, cmd);
+	st7789h2_wrn_pulse(config);
+
+	if (tx_data != NULL) {
+		st7789h2_set_cmd(config, 0);
+		for (size_t count = 0; count < tx_count; count++) {
+			st7789h2_data_bus_set(dev, tx_data[count]);
+			st7789h2_wrn_pulse(config);
+		}
+	}
+	st7789h2_set_cs(config, 0);
+}
+
+static inline void st7789h2_power_up(const struct st7789h2_config *config)
+{
+	if (config->power_gpio.port == NULL) {
+		return;
+	}
+
+	gpio_pin_set_dt(&config->power_gpio, 1);
+}
+
+static inline void st7789h2_power_down(const struct st7789h2_config *config)
+{
+	if (config->power_gpio.port == NULL) {
+		return;
+	}
+
+	gpio_pin_set_dt(&config->power_gpio, 0);
+}
+
+static inline void st7789h2_backlight_on(const struct st7789h2_config *config)
+{
+	if (config->backlight_gpio.port == NULL) {
+		return;
+	}
+
+	gpio_pin_set_dt(&config->backlight_gpio, 1);
+}
+
+static inline void st7789h2_backlight_off(const struct st7789h2_config *config)
+{
+	if (config->backlight_gpio.port == NULL) {
+		return;
+	}
+
+	gpio_pin_set_dt(&config->backlight_gpio, 0);
+}
+
+static void st7789h2_exit_sleep(const struct device *dev)
+{
+	st7789h2_transmit(dev, ST7789H2_CMD_SLEEP_OUT, NULL, 0);
```

Comment:
Cannot `st7789h2_transmit()` subsequent calling `gpio_pin_set_dt()` fail?

Reply:
Not sure I got what you mean, can you be more precise here ?
To enter or exit sleep on the display, only a command through the bus is necessary.